src/components/SingleList.jsx

```diff
@@ -1,19 +1,102 @@
-import { useNavigate } from 'react-router-dom';
 import './SingleList.css';
+import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
+import PushPinIcon from '@mui/icons-material/PushPin';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { MaterialListButton, MaterialButton } from './material-buttons';
+import { useAuth, deleteList } from '../api';
+import { ConfirmDialog } from './ConfirmDialog';
+
+export const deletionResponse = {
+	hard: `List deleted permanently.`,
+	soft: `List removed from user view.`,
+};
+
+export function SingleList({
+	item,
+	setListPath,
+	setImportantList,
+	isImportant,
+}) {
+	const [isHovered, setIsHovered] = useState(false);
+	const [open, isOpen] = useState(false);
 
-export function SingleList({ name, path, setListPath }) {
 	const navigate = useNavigate();
+	const { user } = useAuth();
+	const userId = user?.uid;
+	const userEmail = user?.email;
 
-	function handleClick() {
+	const { name, path } = item;
+	const listPath = path.slice(0, path.indexOf('/'));
+
+	const handleNavigate = () => {
 		setListPath(path);
 		setTimeout(() => {
 			navigate('/list');
 		}, 200);
-	}
+	};
+
+	const handleImportantList = () => {
+		setImportantList(path);
+	};
+
+	const toggleDialog = () => {
+		isOpen((prev) => !prev);
+	};
+
+	const handleDelete = async () => {
+		const deletionType = listPath === userId ? 'hard' : 'soft';
```

Comment:
Group PR review: we didn't test or look at this, but curious to know in what case the `listPath` would exactly match the `userId`. We were wondering if it should be something like string includes instead of `===`.

Reply:
I appreciate this comment!. My variable naming isn't semantic here. In reality the listPath is the user id sliced from the listPath, so I will rename this. 
Thank you!